### PR TITLE
[Inductor] support Conv/Linear + broadcast add fusion

### PR DESCRIPTION
**Pull request by jiayisunx**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #138201

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @gujinghui @PenghuiCheng @jianyuh @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen @snadampal @voznesenskym @penguinwu @EikanWang @zhuhaozhe @blzheng @wenzhe-nrv @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: .gitmodules [hunk 1]
```diff
@@ -61,7 +61,7 @@
 [submodule "third_party/ideep"]
     ignore = dirty
     path = third_party/ideep
-    url = https://github.com/intel/ideep
+    url = https://github.com/jiayisunx/ideep
```
[the remark this answers is not in the document]
Comment (jiayisunx) [reply .gitmodules:64]: This is temporary for CI testing. I will upgrade ideep first and then try to merge the PR.

**Review by jgong5 — CHANGES_REQUESTED** [review]
Please tag me for review after you upgrade ideep and CI passes.

**Comment by jiayisunx** [comment]
> Please tag me for review after you upgrade ideep and CI passes.

It's ready for review, could you please review it again? thanks!

File: third_party/ideep
Comment (jgong5) [inline third_party/ideep:1]: Better to separate this ideep upgrade to a different PR?

**Review by jgong5 — APPROVED** [review]

File: third_party/ideep
Comment (jiayisunx) [reply third_party/ideep:1]: Okay, thanks for your comments!

**Review by jansel — APPROVED** [review]

**Comment by masnesral** [comment]
@jiayisunx / @jgong5 any guesses on why the new tests `test_conv2d_binary_broadcast_shapes_cpu` and `test_conv3d_binary_broadcast_shapes_cpu` would fail with: `RuntimeError: could not create a primitive descriptor for a reorder primitive`. We've got failures on Meta's internal testing. I wonder if it's just a matter of skipping if some lib version is too old?

Note that the failure doesn't seem to be related to torch.compile at all. I see the same problem if I try `torch.ops.mkldnn._convolution_pointwise.binary()` directly in eager..

**Comment by leslie-fang-intel** [comment]
If I remember correctly, this PR has a update of IDEEP version. cc @jiayisunx to take a look.

**Comment by masnesral** [comment]
@jiayisunx thanks for looking. It looks like we have incorporated those 2 PRs internally. Any other ideas that come to mind?

**Comment by jiayisunx** [comment]
> @jiayisunx thanks for looking. It looks like we have incorporated those 2 PRs internally. Any other ideas that come to mind?

hi @masnesral , Is this PR (https://github.com/pytorch/pytorch/pull/141759) included internally? This PR was internally tested by @atalman before being merged and it passed. Are the internal tests you two are talking about the same test? You mentioned that you reproduced the issue by running torch.ops.mkldnn._convolution_pointwise.binary() directly in eager. Can you share this reproducer with me and I will try it locally?

**Pull request closed without merging**